src/components/shared/LikeButton.js

```diff
@@ -0,0 +1,27 @@
+import React, {useEffect, useState} from "react";
+import FavoriteIcon from "@material-ui/icons/Favorite";
+import {addFavoriteAction, removeFromFavoritesAction} from "../../store/reducers/favoritesReducer";
+import {useDispatch, useSelector} from "react-redux";
+
+export function LikeButton({breedId}) {
+    const dispatch = useDispatch();
+
+    const addFavoriteHandler = () => {
+        dispatch(addFavoriteAction(breedId))
+    }
+    const removeFromFavoritesHandler = () => {
+        dispatch(removeFromFavoritesAction(breedId))
+    }
+    const [color, setColor] = useState('primary')
+    const likeDislike = () => {
+        color === 'primary' ? setColor('secondary') : setColor('primary')
+        favoriteIdsState.includes(breedId) ? removeFromFavoritesHandler() : addFavoriteHandler()
```

Comment:
Also try to use if-else construction. Also you don't need to wrapper functions as addFavoriteHandler and removeFromFavoritesHandler, you can use dispatch(action) directly